plugins/compiler.js

```diff
@@ -67,7 +70,7 @@
       return callback(code);
 
     var board = Espruino.Core.Env.getBoardData();
-    var tasks = 0;
+    var tasks = 0; // TODO: This is re-declared as an array at #73, but this outer scoped 'tasks' is not modified
```

Comment:
This definition of `tasks` is a number, lower down there's a test that tests this is `0` or not, but I don't see anywhere this outer scoped `tasks` is modified. The inner scope has a redefinition of `tasks` at :76 as a list of objects.

Reply:
Yes, this should go (or we define it and then set it in the `try` below) - but I believe it's meant to be accessed from replaceNode